addons/mail/static/src/messaging/test_utils.js

```diff
@@ -68,20 +68,82 @@ const MockMailService = Class.extend({
     },
     /**
      * @param {Object} [env={}]
+     * @param {Object} [param1={}]
+     * @param {boolean} [param1.hasTimeControl=false]
+     * @param {Object} [param1.session]
      */
-    messaging(env = {}) {
+    messaging(env = {}, { hasTimeControl = false, session } = {}) {
         const _t = s => s;
         _t.database = {
             parameters: { direction: 'ltr' },
         };
+        if (hasTimeControl) {
+            if (!env.window) {
+                env.window = {};
+            }
+            // list of timeout ids that have timed out.
+            let timedOutIds = [];
+            // key: timeoutId, value: func + remaining duration
+            const timeouts = new Map();
+            Object.assign(env.window, {
+                clearTimeout: id => {
+                    timeouts.delete(id);
+                    timedOutIds = timedOutIds.filter(i => i !== id);
+                },
+                setTimeout: (func, duration) => {
+                    const timeoutId = _.uniqueId('timeout_');
+                    const timeout = {
+                        id: timeoutId,
+                        isTimedOut: false,
+                        func,
+                        duration,
+                    };
+                    timeouts.set(timeoutId, timeout);
+                    if (duration === 0) {
+                        timedOutIds.push(timeoutId);
+                        timeout.isTimedOut = true;
+                    }
+                    return timeoutId;
+                },
+            });
+            if (!env.testUtils) {
+                env.testUtils = {};
```

Comment:
We plan to simplify `start` so that it only mocks `env`, so I would not make it more complicate by returning the functions that change flow of time (and it doesn't help that the making of OWL test env is in the making of the messaging service class at the moment...).

I think that if times are in the `env`, it also make sense to alter flow of time from the `env`, and since it's only in the tests that we want to do that, this makes sense to namespace the functions with `env.testUtils`.

Reply:
ok for now, to reconsider with start change